File: summer2019/week1/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> sFreqs = new HashMap<>();
        populateFrequencyMap(sFreqs, s);
        HashMap<Character, Integer> tFreqs = new HashMap<>();
        populateFrequencyMap(tFreqs, t);
        return sFreqs.equals(tFreqs);
    }
    
    public void populateFrequencyMap(HashMap<Character, Integer> map, String string) {
        char[] charArray = string.toCharArray();
        for (char c : charArray) {
            if (map.containsKey(c)) {
                map.put(c, map.get(c) + 1);
            } else {
                map.put(c, 1);
            }
        }
    }
}

File: spring2020/unit1/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return self.charCounts(s) == self.charCounts(t)

    def charCounts(self, s: str) -> dict:
        d = {}
        for c in s:
            if c in d:
                d[c] = d[c] + 1
            else:
                d[c] = 1
        return d
